backend-lib/src/main/java/cash/z/ecc/android/sdk/internal/model/ProposalUnsafe.kt

```diff
@@ -0,0 +1,29 @@
+package cash.z.ecc.android.sdk.internal.model
+
+import cash.z.wallet.sdk.ffi.ProposalOuterClass.FeeRule
+import cash.z.wallet.sdk.ffi.ProposalOuterClass.Proposal
+
+class ProposalUnsafe(
+    private val inner: Proposal
+) {
+    init {
+        require(inner.feeRule != FeeRule.FeeRuleNotSpecified) {
+            "Fee rule must be specified"
+        }
+    }
+
+    companion object {
+        fun parse(encoded: ByteArray): ProposalUnsafe {
+            val inner = Proposal.parseFrom(encoded)
+            return ProposalUnsafe(inner)
```

Comment:
Despite this being called `ProposalUnsafe`, I expect we'll do most validation in this class because we have access to the parsed protobuf (which can't be used outside `backend-lib` without requiring `sdk-lib` to depend on the protobuf library, and I figure it's better to keep it being a backend-internal detail). The wrapping `Proposal` in `sdk-lib` is wrapping the exposed types here in safe API types like `Zatoshi`.

Reply:
Yes, that is the way we do it with the other models as well.